src/rules/at-each-key-value-single-line/index.js

```diff
@@ -102,19 +110,26 @@ function mapName(mapDecl, mapNamespace) {
 // Returns the parameters of a map-get or map.get call
 // Returns [map variable, key_variable]
 function mapGetParameters(mapGetDecl, mapNamespace) {
-  const pattern = getNamespacedPattern("get\\((.*), ?(.*)\\)", mapNamespace);
+  const pattern = getNamespacedPattern(
+    "get\\((.*), ?(.*)\\)",
+    mapNamespace,
+    mapGetDecl
+  );
   const parts = mapGetDecl.match(new RegExp(pattern));
 
   return [parts[1], parts[2]];
 }
 
-function getNamespacedPattern(pattern, mapNamespace) {
-  switch (mapNamespace) {
-    case "*":
-      return pattern;
-    case null:
-      return `map[-.]${pattern}`;
+function getNamespacedPattern(pattern, namespace, decl) {
+  if (namespace === null) {
+    return `map[-.]${pattern}`;
   }
 
-  return `${mapNamespace}[-.]${pattern}`;
+  if (namespace === "*") {
+    return pattern;
+  }
+
+  return new RegExp(`${namespace}.${pattern}`).test(decl)
+    ? `${namespace}.${pattern}`
+    : `map-${pattern}`;
```

Comment:
I suggest returning a pattern `(${namespace}\.|map-)${pattern}`, which avoids having to test `decl` (and it could work with the default namespace too, by making it be treated as `map`)

Reply:
That is of course much better. I'm not happy with the current implementation anyway.

Reply:
Speeking of that, I think `moduleNamespace()` might actually be nicer if it was returning the default namespace instead of `null` (i.e. the basename of the module URL), and returning `null` instead of `*` when using a module without namespace. This would have 2 benefits:

- the rule code would not need to have a special case for default namespaces (as that would work the same than for an explicit namespace)
- it would be consistent with the way Sass itself treats the namespace for those `@use` rules, making it less confusing for Sass contributors to contribute here with those helpers (for people that don't contribute to Sass itself, they would not know about the inconsistency and so would not care)

Reply:
Sounds good, let's collect some constellations:

```scss
@use 'sass:color'; // namespace 'color'
@use 'my-module'; // namespace 'my-module'
@use 'my-module.scss'; // namespace 'my-module'
@use 'path/to/my-module'; // namespace 'my-module'
@use 'path/to/my-module.scss'; // namespace 'my-module'
```

Am I missing other possibilities?

Reply:
I think this is OK (if you account for the `.sass` extension too)